doc/api-requests.md

```diff
@@ -0,0 +1,156 @@
+
+For shallow requests, `dandidav` does not make any API requests.
+
+Dandiset Top Level
+------------------
+
+> **dandidav path:** `/dandisets/{dandiset_id}/`
+
+For both deep and shallow requests, `dandidav` makes an API request to `/dandisets/{dandiset_id}/`.
+
+Releases Index
+--------------
+
+> **dandidav path:** `/dandisets/{dandiset_id}/releases/`
+
+For deep requests, `dandidav` paginates over the API's `/dandisets/{dandiset_id}/versions/` endpoint.
+
+For shallow requests, `dandidav` does not make any API requests.
+
+Top Level of a Dandiset Version
+-------------------------------
+
+> **dandidav paths:**
+>
+> - `/dandiset/{dandiset_id}/draft/`
+> - `/dandiset/{dandiset_id}/latest/`
+> - `/dandiset/{dandiset_id}/releases/{version_id}/`
+
+For both deep and shallow requests, if the request path is
+`/dandiset/{dandiset_id}/latest/`, an initial request is made to
+`/dandiset/{dandiset_id}/` to get the version ID of the latest published
+version.
+
+For deep requests, `dandidav` makes API requests to the following endpoints:
+
+- `/dandisets/{dandiset_id}/versions/{version_id}/info/`
+- `/dandisets/{dandiset_id}/versions/{version_id}/assets/paths/` (paginated)
+    - For each separate asset returned from this endpoint, `dandidav` makes
+      another request to
+      `/dandisets/{dandiset_id}/versions/{version_id}/assets/{asset_id}/info/`
+      to fetch further details about the asset.
+- `/dandisets/{dandiset_id}/versions/{version_id}/` (to fetch the version
+  metadata so that its size can be reported)
+
+For shallow requests, `dandidav` makes an API request to
+`/dandisets/{dandiset_id}/versions/{version_id}/info/`.
+
+Metadata File
+-------------
+
+> **dandidav paths:**
+>
+> - `/dandiset/{dandiset_id}/draft/dandiset.yaml`
+> - `/dandiset/{dandiset_id}/latest/dandiset.yaml`
+> - `/dandiset/{dandiset_id}/releases/{version_id}/dandiset.yaml`
+
+For both deep and shallow requests, if the request path is
+`/dandiset/{dandiset_id}/latest/`, an initial request is made to
+`/dandiset/{dandiset_id}/` to get the version ID of the latest published
+version.  Then, for both deep and shallow requests, `dandidav` makes an API
+request to `/dandisets/{dandiset_id}/versions/{version_id}/`.
+
+Asset Path
+----------
+
+> **dandidav paths:**
+>
+> - `/dandiset/{dandiset_id}/draft/{path}`
+> - `/dandiset/{dandiset_id}/latest/{path}`
+> - `/dandiset/{dandiset_id}/releases/{version_id}/{path}`
+>
+> Note that any trailing slashes at the end of `path` are ignored and are
+> stripped before passing to the Archive.
+
+For both deep and shallow requests, if the request path is
+`/dandiset/{dandiset_id}/latest/`, an initial request is made to
+`/dandiset/{dandiset_id}/` to get the version ID of the latest published
+version.
+
+Then, for each initial subpath of `path` that ends with either (a) a component
+ending in ".zarr" or ".ngff" (case insensitive) or (b) the end of the path, an
```

Comment:
sorry -- what do you mean by "that ends with either ... or (b) **the end of the path**". Do you mean that it is equals it?

Reply:
Yes, the last subpath is the path in its entirety.